src/utils/oauth2_util.py

```diff
@@ -130,7 +131,7 @@ def is_endpoint_protected(endpoint_request):
 
     if (
         SecuredHttpRequestUrl(
-            path=endpoint_request.path, method=endpoint_request.method
+            path=str(endpoint_request.url_rule), method=endpoint_request.method
         )
```

Comment:
I have to use the `request.url_rule` instead of `.path` as the path doesn't show the endpoint route in the same way as in the map `ENDPOINT_ROLES_MAP`.

Example:
* for method `get_by_id` if you want to get the route for id=1 `path` shows "/chefs/1" instead the `url_role` shows "/chefs/<chef_id>